dashboard/test/models/user_level_test.rb

```diff
@@ -75,74 +118,97 @@ class UserLevelTest < ActiveSupport::TestCase
+              best_result: Activity::MINIMUM_FINISHED_RESULT
+            )
+          end
+        end
+      end
+    end
   end
 
   test "attempted, passed, and perfected scopes for passed result" do
-    UserLevel.create(user: @user, level: @level, attempts: 0, best_result: Activity::MINIMUM_PASS_RESULT)
-
-    assert_equal 1, UserLevel.count
-    assert_equal 1, UserLevel.attempted.count
-    assert_equal 1, UserLevel.passing.count
-    assert_equal 0, UserLevel.perfect.count
+    assert_difference 'UserLevel.count' do
+      assert_difference 'UserLevel.attempted.count' do
+        assert_difference 'UserLevel.passing.count' do
+          assert_no_difference 'UserLevel.perfect.count' do
+            UserLevel.create(
+              user: @user,
+              level: @level,
+              attempts: 0,
+              best_result: Activity::MINIMUM_PASS_RESULT
+            )
+          end
+        end
+      end
+    end
   end
 
   test "attempted, passed, and perfected scopes for perfected result" do
-    UserLevel.create(user: @user, level: @level, attempts: 0, best_result: Activity::MAXIMUM_NONOPTIMAL_RESULT + 1)
-
-    assert_equal 1, UserLevel.count
-    assert_equal 1, UserLevel.attempted.count
-    assert_equal 1, UserLevel.passing.count
-    assert_equal 1, UserLevel.perfect.count
+    assert_difference 'UserLevel.count' do
+      assert_difference 'UserLevel.attempted.count' do
+        assert_difference 'UserLevel.passing.count' do
+          assert_difference 'UserLevel.perfect.count' do
+            UserLevel.create(
+              user: @user,
+              level: @level,
+              attempts: 0,
+              best_result: Activity::MAXIMUM_NONOPTIMAL_RESULT + 1
+            )
+          end
+        end
+      end
+    end
   end
 
   test "unsubmitting should set best result back to attempted" do
-    ul = UserLevel.create(user: @user, level: @level, attempts: 0, submitted: true, best_result: Activity::REVIEW_REJECTED_RESULT)
+    ul = UserLevel.create(
+      user: @user,
+      level: @level,
+      attempts: 0,
+      submitted: true,
+      best_result: Activity::REVIEW_REJECTED_RESULT
+    )
     ul.update! submitted: false
     assert_equal Activity::UNSUBMITTED_RESULT, ul.best_result
   end
 
   test "driver and navigator user levels" do
-    student1 = create :student
-    student2 = create :student
-
-    level = create :level
-    script = create :script
-    driver = create :user_level, user: student1, level: level, script: script
-    navigator = create :user_level, user: student2, level: level, script: script
-
-    driver.navigator_user_levels << navigator
-
-    driver.reload
-    navigator.reload
-
-    assert_equal [navigator], driver.navigator_user_levels
-    assert_equal [driver], navigator.driver_user_levels
+    assert_equal [@navigator_user_level],
+      @driver_user_level.navigator_user_levels
+    assert_equal [@driver_user_level], @navigator_user_level.driver_user_levels
```

Comment:
The setup for this test was refactored to `setup`, so as to be reused in the new tests.